File: 03-Azure/01-04-AI/01_Agentic_SDLC/src/api-py/models/branch.py
from __future__ import annotations

from pydantic import BaseModel


class Branch(BaseModel):
    branchId: int
    headquartersId: int
    name: str
    description: str | None = None
    address: str | None = None
    contactPerson: str | None = None
    email: str | None = None
    phone: str | None = None


class CreateBranchRequest(BaseModel):
    headquartersId: int
    name: str
    description: str | None = None
    address: str | None = None
    contactPerson: str | None = None
    email: str | None = None
    phone: str | None = None


class UpdateBranchRequest(BaseModel):
    headquartersId: int | None = None
    name: str | None = None
    description: str | None = None
    address: str | None = None
    contactPerson: str | None = None
    email: str | None = None
    phone: str | None = None

File: 03-Azure/01-04-AI/01_Agentic_SDLC/src/api-cs/Models/Branch.cs
namespace OctoSupply.Api.Models;

public sealed record Branch(
    int BranchId,
    int HeadquartersId,
    string Name,
    string? Description,
    string? Address,
    string? ContactPerson,
    string? Email,
    string? Phone);

public sealed class CreateBranchRequest
{
    public int HeadquartersId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? Address { get; set; }
    public string? ContactPerson { get; set; }
    public string? Email { get; set; }
    public string? Phone { get; set; }
}

public sealed class UpdateBranchRequest
{
    public int? HeadquartersId { get; set; }
    public string? Name { get; set; }
    public string? Description { get; set; }
    public string? Address { get; set; }
    public string? ContactPerson { get; set; }
    public string? Email { get; set; }
    public string? Phone { get; set; }
}
